feat(angular): deprecate IonicModule (#31270)
driftyco/ionic: docs/angular/testing.md

Tests for lazy loaded Ionic UI components should only be added under the `/lazy` route. This ensures the `IonicModule` is added.

> [!CAUTION]

> > The lazy loaded build and its `IonicModule` are deprecated and will be removed in a future major version. New components should be tested as standalone components (see below). These lazy tests remain to verify that the deprecated build keeps working while it is still supported.

Reviewer: ```suggestion
> The lazy loaded build, including `IonicModule`, is deprecated and will be removed in a future major version. New components should be tested as standalone components (see below). These lazy tests remain to verify that the deprecated build keeps working while it is still supported.
```

Author: https://github.com/ionic-team/ionic-framework/pull/31270/changes/c4d9d74a943926238bdf0b86b391d87746d45d40
